replica-lib/src/accountsdb_repl_server_factory.rs

```diff
@@ -0,0 +1,29 @@
+use {
+    crate::{
+        accountsdb_repl_server::{AccountsDbReplService, AccountsDbReplServiceConfig},
+        replica_accounts_server::ReplicaAccountsServerImpl,
+        replica_confirmed_slots_server::ReplicaSlotConfirmationServerImpl,
+    },
+    crossbeam_channel::Receiver,
+    solana_runtime::bank_forks::BankForks,
+    solana_sdk::clock::Slot,
+    std::sync::{Arc, RwLock},
+};
+
+pub struct AccountsDbReplServerFactory {}
+
+impl AccountsDbReplServerFactory {
+    pub fn build_accountsdb_repl_server(
+        config: AccountsDbReplServiceConfig,
+        confirmed_bank_receiver: Receiver<Slot>,
+        bank_forks: Arc<RwLock<BankForks>>,
+    ) -> AccountsDbReplService {
+        AccountsDbReplService::new(
```

Comment:
This threw me off a bit. We don't really use factory pattern elsewhere and I'm not sure how much value this wrapper adds, but I'm not opposed especially if you had further plans to expand it.

Reply:
This pattern is used to achieve two goals: the AccountsDbReplService does not need to know about the real implementations of ReplicaSlotConfirmationServer and  ReplicaAccountsServer and the caller to instantiate AccountsDbReplService does not need to know about (ReplicaSlotConfirmationServerImpl and ReplicaAccountsServerImpl. So that we could change the implementation freely. As such we could unit test the AccountsDbReplService and code below its stack without a real validator.